barretenberg/cpp/src/barretenberg/stdlib/hash/blake2s/blake2s.cpp

```diff
@@ -33,6 +33,9 @@ namespace bb::stdlib {
  *    f_1 = |
  *           \  0x000...00   otherwise
  *
+ * Note that although blake2s_state is a circuit constant, we use designated functions such as
```

Comment:
move this comment to where you are using `ranged_less_than`. Also might be worth mentioning that the _initial_ blake2s state is circuit-constant.